src/InputField/index.js

```diff
@@ -323,29 +322,88 @@ const InputField = React.forwardRef((props: Props, ref: Ref) => {
     inputMode,
   } = props;
 
-  const forID = id || (label ? React.useMemo(() => randomID("inputFieldID"), []) : undefined);
+  const forID = React.useMemo(() => {
+    if (id) return id;
+    return label ? randomID("inputFieldID") : undefined;
+  }, [id, label]);
+  const [tooltipShown, setTooltipShown] = useState(false);
+  const [tooltipShownHover, setTooltipShownHover] = useState(false);
+  const labelRef = useRef(null);
+  const iconRef = useRef(null);
 
   return (
     <Field
       component={label ? "label" : "div"}
       spaceAfter={spaceAfter}
       htmlFor={label ? forID : undefined}
     >
-      {label && !inlineLabel && <FormLabel label={label} isFilled={!!value} required={required} />}
-      <InputContainer size={size} disabled={disabled} error={error}>
-        {prefix && <Prefix size={size}>{prefix}</Prefix>}
+      {label && !inlineLabel && (
+        <FormLabel
+          inlineLabel={inlineLabel}
+          filled={!!value}
+          required={required}
+          error={!!error}
+          help={!!help}
+          labelRef={labelRef}
+          iconRef={iconRef}
+          onMouseEnter={() => setTooltipShownHover(true)}
+          onMouseLeave={() => setTooltipShownHover(false)}
+        >
+          {label}
+        </FormLabel>
+      )}
+      <InputContainer
+        size={size}
+        disabled={disabled}
+        error={error}
+        labelRef={label ? null : labelRef}
+      >
+        {prefix &&
+          (inlineLabel && (error || help) ? (
+            <Prefix size={size}>
+              {help && !error && (
+                <StyledIconWrapper ref={iconRef}>
+                  <InformationCircle color="secondary" size="small" />
+                </StyledIconWrapper>
+              )}
+              {error && (
+                <StyledIconWrapper ref={iconRef}>
+                  <AlertCircle color="critical" size="small" />
+                </StyledIconWrapper>
+              )}
+            </Prefix>
+          ) : (
+            <Prefix size={size}>{prefix}</Prefix>
+          ))}
         {label && inlineLabel && (
-          <StyledInlineLabel size={size}>
-            <FormLabel label={label} isFilled={!!value} required={required} />
+          <StyledInlineLabel ref={labelRef} size={size}>
+            <FormLabel
+              filled={!!value}
+              required={required}
+              error={!!error}
+              help={!!help}
+              inlineLabel={inlineLabel}
+            >
+              {label}
+            </FormLabel>
           </StyledInlineLabel>
         )}
-        {tags && <InputTags>{tags}</InputTags>}
         <Input
           data-test={dataTest}
           data-state={getFieldDataState(!!error)}
           onChange={onChange}
-          onFocus={onFocus}
-          onBlur={onBlur}
+          onFocus={e => {
```

Comment:
I would use `useCallback` for these binds. When user will be interacting with `InputField`, it's gonna be re-created in every render.